service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/validation/ItkValidator.java

```diff
@@ -44,6 +51,27 @@ public void checkItkConformance(Map<ReportElement, Element> report) throws SoapC
         checkAuditIdentity(itkHeader);
     }
 
+    public void checkItkOdsAndDosId(String odsCode, String dosServiceId) throws SoapClientException {
```

Comment:
this method should accept clean odsCode and dosId as params - it should not parse the string